### ISSUE TITLE
Default mount type and slow VM

### ISSUE THREAD

**Issue by LucasDemea** [roots/trellis#1428]
### Terms

- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This request is not a duplicate of an existing issue
- [X] I have read the [docs](https://roots.io/docs/) and followed them (if applicable)
- [X] I have seached the [Roots Discourse](https://discourse.roots.io/) for answers and followed them (if applicable)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

#### What's wrong?
On a linux host (ubuntu 22.04), my VMs have really bad performance. The TTFB is around 5-6 sec, and navigating the back-end is a pain, often having to wait 10-15 secs for a page to load.
I also noticed some very long durations, sometimes leading to timeouts when doing operation such as `rm -rf vendor`, or 
`composer install` inside the VM. The same operations are at least 10x faster when I run them on the host.

#### What have you tried?
I found that the R/W speeds are bad inside the NFS mounted folders, so I tried the other protocols, by changing the `vagrant_mount_type:` setting in `vagrant.default.yml` to `virtualbox`. 
I compared the writing speeds with `dd if=/dev/zero of=./test bs=1M count=1000` inside one of the shared folders, and found that virtualbox protocol had much better results than NFS:
- 125 MB/s with NFS protocol
- 1.2 GB/s with virtualbox protocol

The TTFB is also reduced, and navigating my sites is much faster.

#### What insights have you gained?
In my case, and possibly on other Linux hosts, the virtualbox shared folders protocol is much faster than NFS.

#### Suggestions
We first need more testing from the community, but `virtualbox` protocol could be suggested as the default trellis protocol.

### Steps To Reproduce

Switch the `vagrant_mount_type` setting from `nfs` to `virtualbox ` and run `vagrant reload`.

### Expected Behavior

NFS should be the fastest protocol

### Actual Behavior

virtualbox is faster than NFS

### Versions

ubuntu 22.04, trellis 1.9.1, vagrant 2.3.0, virtualbox 6.1.38

**Comment by swalkinshaw** [comment]
Interesting!

Here's an older article by the Vagrant creator: https://mitchellh.com/writing/comparing-filesystem-performance-in-virtual-machines

I found https://github.com/winnfsd/vagrant-winnfsd/wiki/Benchmarking-winnfsd-vs-VirtualBox-Shared-Folders and https://github.com/winnfsd/vagrant-winnfsd/issues/46 which has some benchmarks and some good discussion on _how_ to benchmark different types of FS operations.

Both seem to recommend NFS overall, but more specifically when _reads_ are more important which is true for running a WordPress site for example (in normal situations). But for operations like running `composer install`, that's all writes where vboxfs should win out.

Maybe this is different on a Linux host though? 

It would be interesting to trace how many reads + writes are involved in a page load.

**Comment by LucasDemea** [comment]
Thank you for the resources. I will do more in-depth testing when I get the time.

**Comment by LucasDemea** [comment]
I did more testing, using the fio tool this time. 

I used [the recommended tests detailed in this post](https://arstechnica.com/gadgets/2020/02/how-fast-are-your-disks-find-out-the-open-source-way-with-fio/#:~:text=here%20on%20out.-,Ars%20recommended%20tests,4KiB%20random%20write%20process,-fio%20--name%3Drandom) as a base.

I'm not at all an expert in filesystems, and I won't pretend these tests have any scientific value. But I still think the results are interesting and could open the discussion, and maybe someone more knowledgeable could participate.

### Host setup:
Samsung SSD 970 EVO Plus 1TB 
ubuntu 22.10, vagrant 2.3.2, virtualbox 7.0.2, trellis 1.9.1

### Guest setup:
vagrant_cpus: 4
vagrant_memory: 2048

### The results
#### Random write speed (MiB/s)

<table>
<tr>
	<td>
	<td>NFS
	<td>Virtualbox
<tr>
	<td>Single 4KiB random write 
	<td>27.2
	<td>52.1
<tr>
	<td>16 parallel 64KiB random write
	<td>165
	<td>1309
<tr>
	<td>Single 1MiB random write
	<td>109
	<td>993
</table>


#### Random read speed (MiB/s)

<table>
<tr>
	<td>
	<td>NFS
	<td>Virtualbox
<tr>
	<td>Single 4KiB random write 
	<td>9
	<td>66.7
<tr>
	<td>16 parallel 64KiB random write
	<td>64.7
	<td>1653
<tr>
	<td>Single 1MiB random write
	<td>246
	<td>1365
</table>


We can see that virtualbox mount type outperforms (by far) NFS, for both reading & writing.

**Comment by swalkinshaw** [comment]
Wow that's huge. I can't test this on macOS unfortunately since I'm using Parallels for ARM.

**Comment by swalkinshaw** [comment]
@LucasDemea I'm hesitant to make this change the default for all Linux users without some more data points from other users. But in the meantime, would you be interested in adding a section to our [Vagrant docs](https://github.com/roots/docs/blob/docs/docs/trellis/master/vagrant.md) about mount types specifically about Linux + VB?

**Comment by LucasDemea** [comment]
I agree that we need some feedback from other users before making the change.
I'll create a PR to the docs when I get the time.

**Comment by swalkinshaw** [comment]
Closing since docs were added in https://github.com/roots/docs/pull/439

Plus we recommend Lima over Vagrant now
